### ISSUE TITLE
C# Intellisense for VSCode

### ISSUE THREAD

**Issue by rongjiecomputer** [microsoft/BuildXL#386]
It seems like the VSCode plugin built by `bxl -deployconfig release out\bin\release\ide\*` only does DScript language support. The official VSCode C# extension by Microsoft does not understand the files generated by `bxl /vs`.

**Comment by dannyvv** [comment]
Thanks @rongjiecomputer,

I completely agree we need support for that. I have some plans to work in July for a week on getting this working natively in VsCode. The plan will be using a similar pattern to cpp did with: https://github.com/Microsoft/vscode-cpptools-api and provide a direct input from BuildXL graph to the csharp extension

**Comment by rongjiecomputer** [comment]
As I couldn't wait that long for a VSCode plugin, I installed VS2019 IDE and the VS plugin. I am quite sure the plugin is loaded (because VS is complaining that the plugin hurts startup performance), but it doesn't seem to have any effect (can't autocomplete BuildXL's C# code).

**Comment by dannyvv** [comment]
Correct, we have another bug (#163) tracking the update to 2019 format.

**Comment by dannyvv** [comment]
Serge and I implemented the prototype for vscode and cpp in a branch in his repo: https://github.com/smera/BuildXL/tree/hackathon/vscodecpp
I looked at using a similar approach using Omnisharp. But there did not yet exist a loop back mechanism back into vscode from omnisharp. The 2 days were not enough for me to set that up as another ProjectProvider, although I don't see any blockers for it.

I'll mark this item as 'help wanted'  in case someone wants to do the work in OmniSharp to allow code in vscode like the vscode-cpptools-api or via something like [Build Server Protocol](https://github.com/scalacenter/bsp/blob/master/docs/bsp.md) from the Scala community.

Untill then the previously linked workitem should address it by generating the csproj files and have omnisharp pick those up.
